src/components/ProfilePage.jsx

```diff
@@ -0,0 +1,152 @@
+import { useParams } from "react-router-dom";
+import { getRequest, putRequest } from "../utilites/apiRequests";
+import { useEffect, useState } from "react";
+import { ProfileIcon } from "./General/ProfileIcon";
+
+export const ProfilePage = () => {
+	const [user, setUser] = useState(null);
+	const [formData, setFormData] = useState(null);
```

Comment:
Initializing formData with null could potentially lead to errors when trying to access its properties before it's set. Consider initializing it with an empty object or the expected structure with empty values. This can prevent errors if the component tries to render before the data is fetched.